resilience4j-micronaut/src/main/java/io/github/resilience4j/bulkhead/BulkheadSpecificationInterceptor.java

```diff
@@ -0,0 +1,234 @@
+import io.micronaut.retry.intercept.RecoveryInterceptor;
+import io.reactivex.Flowable;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * A {@link MethodInterceptor} that intercepts all method calls which are annotated with a {@link io.github.resilience4j.annotation.Bulkhead}
+ * annotation.
+ **/
+@Singleton
+@Requires(classes = BulkheadRegistry.class)
+public class BulkheadSpecificationInterceptor implements MethodInterceptor<Object,Object> {
+    private static final Logger LOG = LoggerFactory.getLogger(BulkheadSpecificationInterceptor.class);
+
+
+    /**
+     * Positioned before the {@link io.github.resilience4j.annotation.Bulkhead} interceptor after {@link io.micronaut.retry.annotation.Fallback}.
+     */
+    public static final int POSITION = RecoveryInterceptor.POSITION + 20;
+
+    private final BulkheadRegistry bulkheadRegistry;
+    private final BeanContext beanContext;
+
+    /**
+     *
+     * @param beanContext The bean context to allow for DI of class annotated with {@link javax.inject.Inject}.
+     * @param bulkheadRegistry bulkhead registry used to retrieve {@link Bulkhead} by name
+     */
+    public BulkheadSpecificationInterceptor(BeanContext beanContext, BulkheadRegistry bulkheadRegistry) {
+        this.bulkheadRegistry = bulkheadRegistry;
+        this.beanContext = beanContext;
+    }
+
+    @Override
+    public int getOrder() {
+        return POSITION;
+    }
+
+    /**
+     * Finds a fallback method for the given context.
+     *
+     * @param context The context
+     * @return The fallback method if it is present
+     */
+    public Optional<? extends MethodExecutionHandle<?, Object>> findFallbackMethod(MethodInvocationContext<Object, Object> context) {
+        ExecutableMethod executableMethod = context.getExecutableMethod();
+        final String fallbackMethod = executableMethod.stringValue(io.github.resilience4j.annotation.Bulkhead.class, "fallbackMethod").orElse("");
+        Class<?> declaringType = context.getDeclaringType();
+        return beanContext.findExecutionHandle(declaringType, fallbackMethod, context.getArgumentTypes());
+    }
+
+    @Override
+    public Object intercept(MethodInvocationContext<Object, Object> context) {
+
+        Optional<AnnotationValue<io.github.resilience4j.annotation.Bulkhead>> opt = context.findAnnotation(io.github.resilience4j.annotation.Bulkhead.class);
+        if (!opt.isPresent()) {
+            return context.proceed();
+        }
+
+        ExecutableMethod executableMethod = context.getExecutableMethod();
+        final String name = executableMethod.stringValue(io.github.resilience4j.annotation.Bulkhead.class).orElse("default");
+
+        Bulkhead bulkhead = this.bulkheadRegistry.bulkhead(name);
```

Comment:
We must check the type of Bulkhead `Bulkhead.Type.THREADPOOL` and create a ThreadPoolBulkhead if requested.